olive/cache.py

```diff
@@ -32,127 +37,243 @@ class CacheSubDirs:
-        # sometimes the folder is created with contents but the json is not created when the pass fails to run
-        # so we check for both when determining the new model number
-        model_files = list(self.dirs.models.glob("*_*"))
-        if len(model_files) > 0:
-            self.new_model_number = max(int(model_file.stem.split("_")[0]) for model_file in model_files) + 1
+        self.enable_shared_cache = cache_config.enable_shared_cache
+        if self.enable_shared_cache:
+            self.shared_cache = SharedCache(cache_config.account_name, cache_config.container_name)
+        self.update_shared_cache = cache_config.update_shared_cache
+
+    @staticmethod
+    def get_run_json(pass_name: int, pass_config: dict, input_model_id: str, accelerator_spec: "AcceleratorSpec"):
+        accelerator_spec = str(accelerator_spec) if accelerator_spec else None
+        return {
+            "input_model_id": input_model_id,
+            "pass_name": pass_name,
+            "pass_config": pass_config,
+            "accelerator_spec": accelerator_spec,
+        }
 
-    def get_cache_dir(self) -> Path:
-        """Return the cache directory."""
-        return self.cache_dir
+    @classmethod
+    def from_cache_env(cls) -> "OliveCache":
+        """Create an OliveCache object from the cache directory environment variable."""
+        cache_dir = os.environ.get("OLIVE_CACHE_DIR")
+        if cache_dir is None:
+            logger.debug("OLIVE_CACHE_DIR environment variable not set. Using default cache directory.")
+            cache_dir = Path(DEFAULT_CACHE_DIR).resolve() / DEFAULT_WORKFLOW_ID
+        return cls(cache_config={"cache_dir": cache_dir})
 
-    def get_new_model_number(self) -> int:
-        """Get a new model number."""
-        while True:
-            new_model_number = self.new_model_number
-            self.new_model_number += 1
-            if not list(self.dirs.models.glob(f"{new_model_number}_*")):
-                break
-        return new_model_number
+    def cache_model(self, model_id: str, model_json: Dict):
+        model_json_path = self.get_model_json_path(model_id)
+        try:
+            with model_json_path.open("w") as f:
+                json.dump(model_json, f, indent=4)
+            logger.debug("Cached model %s to %s", model_id, model_json_path)
 
-    def get_model_json_path(self, model_id: str) -> Path:
-        """Get the path to the model json file."""
-        return self.dirs.models / f"{model_id}.json"
+        except Exception:
+            logger.exception("Failed to cache model to local cache.")
 
-    def get_model_output_path(self, model_id: str) -> Path:
-        """Get the path to the model output directory."""
-        output_path = self.dirs.models / model_id / "output_model"
-        output_path.mkdir(parents=True, exist_ok=True)
-        return output_path
+        if self.enable_shared_cache and self.update_shared_cache:
```

Comment:
Should we check for `self.enable_shared_cache`??
Without it you could support "upload only" scenario i.e. "no downloads but still update the remote cache".

Reply:
The default value of `update_shared_cache` could be a problem. Right now `enable_shared_cache` is not needed for users if they don't want to enalbe shared cache, if `update_shared_cache` always be True, `enable_shared_cache` will always be needed for users.